### PR TITLE
Raise an exception if the input message is not a valid message representative

### LINKED ISSUE #84: Move MGF and Hash Functions to parameters, not inputs
Spec currently uses MGF and Hash Function as inputs. Would be better to have those be parameters.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #84 and #85.

### CODE REVIEW

File: draft-irtf-cfrg-rsa-blind-signatures.md [hunk 1]
```diff
@@ -270,14 +271,17 @@ Outputs:
 Errors:
 - "unexpected input size": Raised when a byte string input doesn't
   have the expected length.
+- "invalid message length": Raised when the message representative
+  to sign is not an integer between 0 and n - 1
```
Comment (chris-wood) [inline draft-irtf-cfrg-rsa-blind-signatures.md:275]:
```suggestion
  to sign is not an integer between 0 and n - 1.
```

File: draft-irtf-cfrg-rsa-blind-signatures.md [hunk 2]
```diff
@@ -270,14 +271,17 @@ Outputs:
...
+  to sign is not an integer between 0 and n - 1.
 
 Steps:
 1. If len(blinded_msg) != kLen, raise "unexpected input size"
    and stop
 2. m = OS2IP(blinded_msg)
-3. s = RSASP1(skS, m)
-4. blind_sig = I2OSP(s, kLen)
-5. output blind_sig
+3. If m < n, raise "invalid message length" and stop
```
Comment (jedisct1) [inline draft-irtf-cfrg-rsa-blind-signatures.md:281]: Did you mean m >= n ?

**Review by jedisct1 — CHANGES_REQUESTED** [review]

File: draft-irtf-cfrg-rsa-blind-signatures.md (hunk 2 above)
Comment (chris-wood) [reply draft-irtf-cfrg-rsa-blind-signatures.md:281]: Hah, oops, yes. I didn't return an error before but changed it without the check.

Comment (chris-wood) [reply draft-irtf-cfrg-rsa-blind-signatures.md:281]:
```suggestion
3. If m >= n, raise "invalid message length" and stop
```

**Review by jedisct1 — APPROVED** [review]

**Comment by FredericJacobs** [comment]
👍👍👍

**Pull request merged**